src/zkevm_specs/util/param.py

```diff
@@ -72,6 +74,8 @@
 MEMORY_EXPANSION_QUAD_DENOMINATOR = 512
 # Coefficient of linear part of memory expansion gas cost
 MEMORY_EXPANSION_LINEAR_COEFF = 3
+# Coefficient of linear part of memory expansion gas cost, specifically in the case of SHA3
+MEMORY_EXPANSION_LINEAR_COEFF_SHA3 = 6
```

Comment:
Do you have the source where this parameter comes from? I don't think SHA3 has a special memory expansion cost.

Reply:
So I initially checked here: https://evm.codes

Then verified it here: https://github.com/ethereum/go-ethereum/blob/master/core/vm/gas_table.go#L259

which uses: https://github.com/ethereum/go-ethereum/blob/master/params/protocol_params.go#L40

Reply:
Right, but we have two variables in `param.py`, `GAS_COST_COPY_SHA3` and `MEMORY_EXPANSION_LINEAR_COEFF_SHA3`. They both have the value 6.
It looks like the `Keccak256WordGas=6` from `protocol_params.go` works like `GAS_COST_COPY_SHA3`.

But I don't get the `MEMORY_EXPANSION_LINEAR_COEFF_SHA3` one. You use it in the test with `memory_expansion`.
I'm also wondering why the test didn't fail if I change the `memory_expansion` back using the 3 gas expansion cost.

Reply:
I declared the `MEMORY_EXPANSION_LINEAR_COEFF_SHA3` only because there was a `MEMORY_EXPANSION_LINEAR_COEFF` that was being used. So I assumed I was supposed to follow the same pattern.

What change did you make specifically before testing?

Reply:
I made this change
```diff
     next_memory_size, memory_expansion_cost = memory_expansion(
-        offset + length, offset + length, MEMORY_EXPANSION_LINEAR_COEFF_SHA3
+        offset + length, offset + length
     )
```

Reply:
Oh nice catch. So the `MEMORY_EXPANSION_LINEAR_COEFF` should stay at `3`. It's only the multiplier for the minimum word size that changes from `3` to `6` (in the `SHA3` case).

The reason why the tests still passed after you made the above change is that, since the memory was already populated until `offset + length`, there is no further expansion and hence the the `memory_expansion_cost` is `0`.

Reply:
Fixed it in the latest commit :)